File: springboot/src/main/java/com/hypermedicus/service/AccountService.java
package com.hypermedicus.service;

import com.hypermedicus.common.utils.JWTUtil;
import com.hypermedicus.connect.AccountConnect;
import com.hypermedicus.mapper.AccountMapper;
import com.hypermedicus.model.AccountDTO;
import com.hypermedicus.payload.response.AccountResponse;
import com.hypermedicus.rds.entity.AccountEntity;
import com.hypermedicus.rds.entity.AccountInformationEntity;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.UUID;

@Service
public class AccountService {

    private final AccountMapper mapper;
    private final AccountConnect accountConnect;

    public AccountService(AccountMapper mapper
    , AccountConnect accountConnect) {
        this.mapper = mapper;
        this.accountConnect = accountConnect;
    }

    @Transactional
    public AccountResponse signUpAccount(AccountDTO dto) {
        UUID uuid = UUID.randomUUID();
        dto.setUuid(uuid);

        AccountEntity accountEntity = mapper.toAccountEntity(dto, uuid);
        AccountInformationEntity accountInformationEntity = mapper.toAccountInformationEntity(dto, uuid);

        accountConnect.signUpAccount(accountEntity, accountInformationEntity); 

        AccountResponse accountResponse = accountConnect.getAccount(uuid);
        
        String accessToken = JWTUtil.generateToken(uuid, 10);
        String refreshToken = JWTUtil.generateToken(uuid, 60*24);

        accountResponse.setAccessToken(accessToken);
        accountResponse.setRefreshToken(refreshToken);
        
        return accountResponse;
    }
}

File: fastapi/app/service/AccountService.py
from app.model.AccountDTO import AccountDTO
from app.common.utils.JWTUtil import JWTUtil

class AccountService:
    @staticmethod
    def signUpAccount(in_data: AccountDTO):
        import uuid
        from app.connect.AccountConnect import AccountConnect
        create_uuid: uuid = uuid.uuid4()
        result = AccountConnect.signUpAccount(in_data, create_uuid)

        if result is not None:            
            account, information = result
            account_dict = {column.name: getattr(account, column.name) for column in account.__table__.columns}
            information_dict = {column.name: getattr(information, column.name) for column in information.__table__.columns}            
                
            account_dict.pop("uuid")
            information_dict.pop("uuid")

            access_token = JWTUtil.generateToken(create_uuid, 10)            
            refresh_token = JWTUtil.generateToken(create_uuid, 60 * 24)

            information_dict["refresh_token"] = refresh_token

            result_dict = {
                "account": account_dict,
                "accountinformation": information_dict,
                "access_token": access_token
            }
            return result_dict
        return result
